### DISCUSSION TITLE
Feature Idea: Automatic indentation of chapters

### DISCUSSION THREAD

**Discussion by leo-hstone** [Zettlr/Zettlr#2932] [Ideas]
In my experience of using zettlr I am having difficulties of seeing instantly to which chapter/heading a line of text belongs. I guess this is mostly due to only size differences in heading formatting. I would love to have the option to automatically ident chapters in regard to heading level. 

The output could look something like this:
![image](https://user-images.githubusercontent.com/62036826/146361197-c7895c25-5d98-427a-b707-af4b262f2f37.png)

As headings are already detected (to be collapsable), I hope this feature is not impossible to implement. I would love to contribute, but unfortunately I am not experienced enough. 

Was this idea already proposed? What do you think?

**Comment by leo-hstone** [comment]
I just found a similar [post ](https://forum.zettlr.com/discussion/394/cascade-indentation-in-heading-levels) in the old forum - sorry for the duplicate.

Hendrik wrote there, that this feature is not conform with the specification of markdown. As far as I understand the specification is only relevant for the actual markup file. Writing this idea, I had not in mind to actually insert any characters to the file but rather just display the relevant sections indented. Is this possible?

**Reply by nathanlesage** [reply]
Changing this whole appearance is in theory possible, but requires extensive changes to the underlying code base, so I'm not sure it would be worth the hassle.
